### DISCUSSION TITLE
Does CoreWebView2Environment.CreateWebResourceResponse dispose the content stream

### DISCUSSION THREAD

**Discussion by fvanheeswijk** [MicrosoftEdge/WebView2Feedback#3873] [Q&A]
Hello,

We are using [CoreWebView2Environment.CreateWebResourceResponse](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2environment.createwebresourceresponse?view=webview2-dotnet-1.0.818.41) and are unsure whether this method disposes the stream itself when the response is finished or if it for example creates a copy of the stream.

So our question is, are we allowed to dispose the content stream after passing it to this method?

**Comment by fw2568** [comment]
Good question @fvanheeswijk, have you ever found out? At least a direct file stream stays open after passing it to the method, so I would say it is never disposed.

**Comment by trametheka** [comment]
Bump, also interested in this. Have confirmed that streams given to CoreWebView2WebResourceRequest & CoreWebView2WebResourceResponse are not disposed, and there doesn't seem to be a way to tell when the class is finished with the stream (happy to be pointed to what I've missed). 

Is there a best practice for when a WebResource's stream should be disposed?
